public/app/plugins/datasource/cloudwatch/datasource.js

```diff
@@ -379,7 +389,7 @@ function (angular, _, moment, dateMath, kbn, CloudWatchAnnotationQuery) {
 
         if (dimensionKey) {
           var variable = _.find(templateSrv.variables, function(variable) {
-            return templateSrv.containsVariable(target.dimensions[dimensionKey], variable.name);
+            return self.containsVariable(target.dimensions[dimensionKey], variable.name);
```

Comment:
that is true, it's here https://github.com/grafana/grafana/blob/46d4f817e3f6e0e7ca39027b155d084d14dd2d8a/public/app/features/templating/variable.ts#L23 

But why the contains check? templating is for interpolation. But for cloudwatch it is not used for interpolation, but looking up the variable and it's values, so why not just look up the variable by name check? no need to use contains